Swiftz/JSON.swift

```diff
@@ -21,32 +21,30 @@ public enum JSONValue : CustomStringConvertible {
 			return NSArray(array: xs.map { $0.values() })
 		case let .JSONObject(xs):
 			return Dictionary.fromList(xs.map({ k, v in
-				return (NSString(string: k), v.values())
+				return (k, v.values())
 			}))
 		case let .JSONNumber(n):
 			return n
 		case let .JSONString(s):
-			return NSString(string: s)
-		case .JSONNull():
+			return s
+		case .JSONNull:
 			return NSNull()
 		}
 	}
 
 	// we know this is safe because of the NSJSONSerialization docs
-	private static func make(a : NSObject) -> JSONValue {
+	public static func make(a : AnyObject) -> JSONValue {
 		switch a {
```

Comment:
I'll make it private it again, but will work on an implementation that is typesafe in all cases and allows returning a proper error. I'm thinking something similar to Aeson's eitherDecode using an Either<String?, JSONDecodingError> sort of type.